java/ql/src/semmle/code/java/frameworks/spring/SpringRestTemplate.qll

```diff
@@ -0,0 +1,93 @@
+import java
+
+import semmle.code.java.frameworks.spring.SpringWebClient
+
+/**Models `RestTemplate` class's `doExecute` method */
+class RestTemplateDoExecute extends Method {
```

Comment:
All code relating to these (as noted above I think we can factor a lot of it) should be moved to the existing file `SpringWebClient.qll`